### ISSUE TITLE
Remove Certain sparc-dev Modules

### ISSUE THREAD

**Issue by jmgate** [trilinos/Trilinos#7504]
## Bug Report
@bartlettroscoe, @e10harvey

### Description
After the transition to the `sparc-dev` modules, sourcing the `load_matching_env.sh` that Trilinos drops as part of the install winds up loading SPARC modules that have nothing to do with Trilinos.  The concrete example I ran into was vvtest, which caused problems for us because it's the wrong version for EMPIRE.  We've already fixed things on our end to get rid of SPARC's vvtest and load our own, but it'd be nice to make sure the Trilinos configuration scripts aren't loading extraneous stuff that's not needed by Trilinos.

### Steps to Reproduce
```bash
$ ssh eclipse.sandia.gov
$ cd /projects/empire/installs/eclipse/INTEL-18.0.2_OPENMPI-4.0.3-RELEASE-OPENMP-STATIC/trilinos/latest
$ source load_matching_env.sh
Hostname 'eclipse-login6' matches known ATDM host 'eclipse-login6' and system 'cts1'
Setting compiler and build options for build-name 'EMPIRE-deploy-Trilinos-eclipse-intel-opt-openmp'
Using cts1 toss3 compiler stack INTEL-18.0.2_OPENMPI-4.0.3 to build RELEASE code with Kokkos node type OPENMP
This version of Open MPI 4.0.3 is temporary and was built with Intel/18.0 compilers and cudatoolkit/10.1


The following have been reloaded with a version change:
  1) intel/19.0 => intel/18.0.2.199
$ which vvtest
/projects/sparc/tools/vvt/vvtest
```

**Comment by bartlettroscoe** [comment]
@e10harvey, it would be good if we could have a genetic bash function:

```
atdm_remove_from_path  /projects/sparc/tools/vvt/vvtest
```

so this sort of thing will be easier in the future.

**Comment by bartlettroscoe** [comment]
@jmgate, this is unfortunately one of the consequences of switching to using the `sparc-dev` modules.  That change will make it much easier to setup and maintain ATDM Trilinos configurations based on the SPARC TPLs but we will have to tweak a few things after we load the `sparc-dev` module.  For example, we had to fix the OpenMP settings in PR #7488.  But this is a small price to pay I think and a good tradeoff.
